### PR TITLE
[minigraph.py] Update minigraph parsing logic to include only active ports for mirror tables

### PR DESCRIPTION
**Pull request by daall**
[minigraph.py] Update minigraph parsing logic to include only active ports for mirror tables

- Update minigraph.py to filter out front-panel ports that are not active
- Update cfggen tests to reflect new behavior

Signed-off-by: Danny Allen <daall@microsoft.com>

**- What I did**
Described above.

**- How I did it**
Described above.

**- How to verify it**
See tests.

**- Description for the changelog**

Update minigraph parsing logic to include only active ports for mirror tables

**- A picture of a cute animal (not mandatory but encouraged)**
cute animal

### CODE REVIEW

File: src/sonic-config-engine/tests/test_cfggen.py [hunk 1]
```diff
@@ -98,11 +98,11 @@ def test_minigraph_acl(self):
...
                                          "{'DATAACL': {'type': 'L3', 'policy_desc': 'DATAACL', 'ports': ['PortChannel01', 'PortChannel02', 'PortChannel03', 'PortChannel04']}, "
                                          "'NTP_ACL': {'services': ['NTP'], 'type': 'CTRLPLANE', 'policy_desc': 'NTP_ACL'}, "
-                                         "'EVERFLOW': {'type': 'MIRROR', 'policy_desc': 'EVERFLOW', 'ports': ['PortChannel01', 'PortChannel02', 'PortChannel03', 'PortChannel04', 'Ethernet24', 'Ethernet40', 'Ethernet20', 'Ethernet44', 'Ethernet48', 'Ethernet28', 'Ethernet96', 'Ethernet92', 'Ethernet76', 'Ethernet72', 'Ethernet52', 'Ethernet80', 'Ethernet56', 'Ethernet32', 'Ethernet16', 'Ethernet36', 'Ethernet12', 'Ethernet60', 'Ethernet8', 'Ethernet4', 'Ethernet0', 'Ethernet64', 'Ethernet68', 'Ethernet84', 'Ethernet88', 'Ethernet108', 'Ethernet104', 'Ethernet100']}, "
+                                         "'EVERFLOW': {'type': 'MIRROR', 'policy_desc': 'EVERFLOW', 'ports': ['PortChannel01', 'PortChannel02', 'PortChannel03', 'PortChannel04']}, "
```
Comment (prsunny) [inline src/sonic-config-engine/tests/test_cfggen.py]: I thought the test was to check with *some* ports removed but not *all*.  Why do you remove all front-panel ports from test?

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -729,6 +731,21 @@ def parse_xml(filename, platform=None, port_config_file=None):
     results['NTP_SERVER'] = dict((item, {}) for item in ntp_servers)
     results['TACPLUS_SERVER'] = dict((item, {'priority': '1', 'tcp_port': '49'}) for item in tacacs_servers)
 
+    # Filter ports from mirroring ACL tables that are not active.
+    for acl_table, group_params in acls.iteritems():
```
Comment (prsunny) [inline src/sonic-config-engine/minigraph.py]: Should this be a separate function? Just a suggestion!.

File: src/sonic-config-engine/minigraph.py [hunk 3]
```diff
@@ -729,6 +731,21 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
+        group_type = group_params.get('type', None)
+
+        if group_type != 'MIRROR' and group_type != 'MIRRORV6':
+            continue
+
+        # We assume that if a LAG was included in the list of ports earlier then it will be active.
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: i am not sure I understand the comment here.

Comment (daall) [reply src/sonic-config-engine/minigraph.py]: The acls object gets populated with data from the minigraph in an earlier step. At this point, that object contains every front panel port, unless it is in a port channel, in which case the port channel is included instead. We make the assumption here that if there is a port channel in the acls object then that port channel will be an active port and should not be filtered out of the MIRROR tables.

**Comment by renukamanavalan** [comment]
Dropped 'request for configlet' as could not be cherry-picked onto 201811.
A new PR created for 201811 -- #3634

**Pull request merged**